src/measure_theory/hausdorff_measure.lean

```diff
@@ -0,0 +1,474 @@
+`measure_theory.outer_measure.mk_metric'`/`measure_theory.outer_measure.mk_metric` to define
+`measure_theory.measure.mk_metric'`/`measure_theory.measure.mk_metric`. We also restate some lemmas
+about metric outer measures for metric measures.
+-/
+
+namespace measure
+
+variables [measurable_space X] [borel_space X]
+
+/-- Given a function `m : set X → ℝ≥0∞`, `mk_metric' m` is the supremum of `μ r`
+over `r > 0`, where `μ r` is the maximal outer measure `μ` such that `μ s ≤ m s`
+for all `s`. While each `μ r` is an *outer* measure, the supremum is a measure. -/
+def mk_metric' (m : set X → ℝ≥0∞) : measure X :=
+(outer_measure.mk_metric' m).to_measure (outer_measure.mk_metric'_is_metric _).le_caratheodory
+
+/-- Given a function `m : ℝ≥0∞ → ℝ≥0∞`, `mk_metric m` is the supremum of `μ r` over `r > 0`, where
+`μ r` is the maximal outer measure `μ` such that `μ s ≤ m s` for all sets `s` that contain at least
+two points. While each `mk_metric'.pre` is an *outer* measure, the supremum is a measure. -/
+def mk_metric (m : ℝ≥0∞ → ℝ≥0∞) : measure X :=
+(outer_measure.mk_metric m).to_measure (outer_measure.mk_metric'_is_metric _).le_caratheodory
+
+@[simp] lemma mk_metric'_to_outer_measure (m : set X → ℝ≥0∞) :
+  (mk_metric' m).to_outer_measure = (outer_measure.mk_metric' m).trim :=
+rfl
+
+@[simp] lemma mk_metric_to_outer_measure (m : ℝ≥0∞ → ℝ≥0∞) :
+  (mk_metric m : measure X).to_outer_measure = outer_measure.mk_metric m :=
+outer_measure.trim_mk_metric m
+
+end measure
+
+lemma outer_measure.coe_mk_metric [measurable_space X] [borel_space X] (m : ℝ≥0∞ → ℝ≥0∞) :
+  ⇑(outer_measure.mk_metric m : outer_measure X) = measure.mk_metric m :=
+by rw [← measure.mk_metric_to_outer_measure, coe_to_outer_measure]
+
+namespace measure
+
+variables [measurable_space X] [borel_space X]
+
+/-- If `c ∉ {0, ∞}` and `m₁ d ≤ c * m₂ d` for `0 < d < ε` for some `ε > 0`
+(we use `≤ᶠ[𝓝[Ioi 0]]` to state this), then `mk_metric m₁ hm₁ ≤ c • mk_metric m₂ hm₂`. -/
+lemma mk_metric_mono_smul {m₁ m₂ : ℝ≥0∞ → ℝ≥0∞} {c : ℝ≥0∞} (hc : c ≠ ∞) (h0 : c ≠ 0)
+  (hle : m₁ ≤ᶠ[𝓝[Ioi 0] 0] c • m₂) :
+  (mk_metric m₁ : measure X) ≤ c • mk_metric m₂ :=
+begin
+  intros s hs,
+  rw [← outer_measure.coe_mk_metric, coe_smul, ← outer_measure.coe_mk_metric],
+  exact outer_measure.mk_metric_mono_smul hc h0 hle s
+end
+
+/-- If `m₁ d ≤ m₂ d` for `0 < d < ε` for some `ε > 0` (we use `≤ᶠ[𝓝[Ioi 0]]` to state this), then
+`mk_metric m₁ hm₁ ≤ mk_metric m₂ hm₂`-/
+lemma mk_metric_mono {m₁ m₂ : ℝ≥0∞ → ℝ≥0∞} (hle : m₁ ≤ᶠ[𝓝[Ioi 0] 0] m₂) :
+  (mk_metric m₁ : measure X) ≤ mk_metric m₂ :=
+by { convert mk_metric_mono_smul ennreal.one_ne_top ennreal.zero_lt_one.ne' _; simp * }
+
+/-!
+### Hausdorff measure and Hausdorff dimension
+-/
+
+/-- Hausdorff measure on an (e)metric space. -/
+def hausdorff_measure (d : ℝ) : measure X := mk_metric (λ r, r ^ d)
+
+localized "notation `μH[` d `]` := measure_theory.measure.hausdorff_measure d" in hausdorff_measure
```

Comment:
```suggestion
localized "notation `μH[` d `]` := measure_theory.measure.hausdorff_measure d" in measure_theory
```
No need to have thousands of locales.

Reply:
This will be the first localized notation in `measure_theory`. Should I move some of current notation to a locale, or make `μH[d]` a global notation, or let `μH[d]` be the first localized notation in `measure_theory`?

Reply:
There are already the notations for L^1 and L^2 in this locale (as of yesterday's master :-)